### PR TITLE
i1714: New tests

### PR DESCRIPTION
**Pull request by MartinEden**
https://vimc.myjetbrains.com/youtrack/issue/VIMC-1714

First review #24.

### CODE REVIEW

**Comment by richfitz** [comment]
https://github.com/vimc/bb8/compare/i1714_restructure...i1714

File: tests/mocks.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def mock_target(id, backup=True, restore=True, **kwargs):
+    return Dynamic(id, id=id, mount_id="mount-{}".format(id),
+                   name="name-{}".format(id),
+                   options=Dynamic("opt", backup=backup, restore=restore),
+                   **kwargs)
+
+
+mock_starport_settings = {
```
Comment (richfitz) [inline tests/mocks.py:18]: should these not be starwars themed?  We're mixing canons here :joy: 
superman-star-wars-1038127

File: tests/test_backup_and_restore.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    ]
+    settings = mock_settings(targets)
+    rsync = Dynamic("rsync", backup_volume=MagicMock())
+
+    # Test
+    run_backup(lambda: settings, rsync)
+
+    # Check
+    rsync.backup_volume.assert_has_calls([
+        call(settings, "mount-a"),
```
Comment (richfitz) [inline tests/test_backup_and_restore.py:23]: I'm totally unfamiliar with this is framework - is this checking that only these two calls happen (i.e., that "b" is not backed up) or that _at least_ these two calls happen?

Comment (richfitz) [reply tests/test_backup_and_restore.py:23]: ok, just confirmed that this is the actual list of calls

File: tests/test_settings.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        "user": "astronaut",
+        "backup_location": "houston"
+    }
+    assert s.targets[0] == NamedVolumeTarget("target_1", "volume_name",
+                                             TargetOptions(True, True))
+    assert s.targets[1] == DirectoryTarget("target_2", "/some/path",
+                                           TargetOptions(False, False))
+
+
+def teardown_module(module):
```
Comment (richfitz) [inline tests/test_settings.py]: If this is created within a test, should it not be torn down in that test too (or set up at the module level?)

**Review by richfitz — CHANGES_REQUESTED** [review]
Can you deal with the merge conflicts, otherwise this is nice.  The mocking will take me a bit of time to get my head around but I can see kind of how it's all working (and have tested locally)

**Comment by hillalex** [comment]
Awesome :)

**Review by hillalex — APPROVED** [review]

File: tests/test_settings.py (hunk 3 above)
Comment (MartinEden) [reply tests/test_settings.py]: Moved the setup to the module level. If we add more tests to this module we may have to rethink.

Comment (richfitz) [reply tests/test_settings.py]: Cool, this is symmetric which is nice

**Review by richfitz — APPROVED** [review]

**Pull request merged**